lib/apiservers/service/restapi/handlers/vch_list_get.go

```diff
@@ -139,3 +145,26 @@ func (h *vchListGet) vchsToModels(op trace.Operation, c *client.HandlerClient, v
 
 	return payload
 }
+
+func (h *vchListGet) parent(op trace.Operation, vch *vm.VirtualMachine) *models.ManagedObject {
+	id := vch.Reference().Value
+
+	var mvm mo.VirtualMachine
+	if err := vch.Properties(op, vch.Reference(), []string{"parentVApp", "resourcePool"}, &mvm); err != nil {
```

Comment:
I'm finding myself wanting to see if there's a way to use the `vm.Parent(...)` call and then check after if the result is a vApp.
Just because it makes it obvious when we come to remove vApp from these code paths that this entire function can go away.